cirq/_compat.py

```diff
@@ -38,7 +38,7 @@ def proper_repr(value: Any) -> str:
         return result
 
     if isinstance(value, np.ndarray):
-        return 'np.array({!r}, dtype=np.{})'.format(value.tolist(), value.dtype)
+        return 'np.array({!r}, dtype=np.{!r})'.format(value.tolist(), value.dtype)
```

Comment:
https://github.com/quantumlib/Cirq/issues/3582

Reply:
We need to make a decision:

1) keep this change to !r which may change some reprs; which could theoretically break someone's code
2) manually edit the json_test_data/*.repr file for BitstringAccumulator to make it parsable. Remove the `assert_equivalent_reprs` test and concede  that this class can't be serialized to repr
3) Really give up on repr 1:1-itude and use the `@dataclass` autogenerated repr which would omit the `cirq.work.` prefix and would use the vanilla numpy repr which truncates large arrays

I personally think 3 is the way forward but it's a big departure for cirq. Repr should give you a good idea of the structure of an object and I don't think we can do that by printing out massive bit arrays

Reply:
[Craig's comment](https://github.com/quantumlib/Cirq/issues/3582#issuecomment-741483678) is a 4th option which I would follow here as well - i.e special casing for np.datetime